src/zerocoin_v3.cpp

```diff
@@ -342,8 +385,15 @@ bool CheckZerocoinTransactionV3(
     // Check Spend Zerocoin Transaction
     if(tx.IsZerocoinSpendV3()) {
         // First check number of inputs does not exceed transaction limit
-        if (tx.vin.size() > consensus.nMaxSigmaSpendPerBlock) {
-            return false;
+        if (tx.vin.size() > consensus.nMaxAmountSigmaSpendPerBlock) {
+            return state.DoS(100, false,
+                REJECT_INVALID,
+                "CheckZerocoinTransactionV3: amount of spends is exceed limit");
+        }
+        if (GetSpendAmount(tx) > consensus.nMaxValueSigmaSpendPerBlock) {
```

Comment:
Call to `GetSpendAmount` possible to throw exception. So better to wrap it in `try`.

Reply:
after discussion handle in `GetSpendAmount` and threat amount as 0.